config/routes.rb

```diff
@@ -11,6 +11,9 @@
       member do
         get '(page-:page)', action: :show, as: '', constraints: page_constraint
       end
+      collection do
+        put :mark_all_read
```

Comment:
If we're going to be shallow and pedantic I try to not use the collection/member router stuff and stick with resources as much as possible.

```ruby
resources :private_topics, except: [:new, :show], path: '' do
  resources :read_state, only: [:destroy]
end
```